runelite-client/src/main/java/net/runelite/client/plugins/prayerbar/PrayerBarOverlay.java

```diff
@@ -0,0 +1,133 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.prayerbar;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Player;
+import net.runelite.api.Point;
+import net.runelite.api.Prayer;
+import net.runelite.api.Skill;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.ui.overlay.*;
+
+@Singleton
+public class PrayerBarOverlay extends Overlay
+{
+	private final Client client;
+	private final PrayerBarConfig config;
+	private final Dimension prayerBarSize = new Dimension(30, 5); // 30x5 is the size of health bars
+	private Player localPlayer;
+	private boolean showPrayerBar = true;
+
+	@Inject
+	private PrayerBarOverlay(PrayerBarConfig config, Client client)
+	{
+		this.config = config;
+		this.client = client;
+
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.LOW);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
+	}
+
+	public void checkToShowPrayerBar()
+	{
+		showPrayerBar = true;
+
+		localPlayer = client.getLocalPlayer();
+
+		if (localPlayer == null)
+		{
+			showPrayerBar = false;
+			return;
+		}
+
+		if (config.hideIfNotPraying() && !isAnyPrayerActive())
+		{
+			showPrayerBar = false;
+			return;
+		}
+
+		if (config.hideIfOutOfCombat() && localPlayer.getHealth() == -1)
+		{
+			showPrayerBar = false;
+		}
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (showPrayerBar)
+		{
+			return renderPrayerBar(graphics, localPlayer);
+		}
+
+		return null;
+	}
+
+	private Dimension renderPrayerBar(Graphics2D graphics, Player localPlayer)
+	{
+		int height = localPlayer.getLogicalHeight() + 15;
+		LocalPoint localLocation = localPlayer.getLocalLocation();
+		Point canvasPoint = Perspective.worldToCanvas(client, localLocation.getX(), localLocation.getY(), client.getPlane(), height);
+
+		//Draw bar
+		int barX = canvasPoint.getX() - 15; // (-15, 3) is to line it up with your health bar, 1 pixel beneath it.
+		int barY = canvasPoint.getY() + 3;
+
+		int barWidth = prayerBarSize.width;
+		int barHeight = prayerBarSize.height;
+
+		float ratio = (float) client.getBoostedSkillLevel(Skill.PRAYER) / client.getRealSkillLevel(Skill.PRAYER);
+
+		int progressFill = (int) Math.ceil(Math.min((barWidth * ratio), barWidth)); // Restricted by the width to prevent the bar from being too long while you are boosted above your real prayer level.
+
+		graphics.setColor(Color.white);
+		graphics.fillRect(barX, barY, barWidth, barHeight);
+		graphics.setColor(Color.cyan);
```

Comment:
Make this Color.white and Color.cyan constants in this class and name them something like BAR_BG_COLOR and BAR_FG_COLOR